crates/nu-command/src/commands/which_.rs

```diff
@@ -214,26 +213,23 @@ fn which_single(application: Tagged<String>, all: bool, scope: &Scope) -> Vec<Va
     }
 }
 
-fn which(args: CommandArgs) -> Result<ActionStream, ShellError> {
+fn which(args: CommandArgs) -> Result<OutputStream, ShellError> {
     let scope = args.scope.clone();
+    let args = args.evaluate_once()?;
 
-    let (
-        WhichArgs {
-            application,
-            rest,
-            all,
-        },
-        _,
-    ) = args.process()?;
+    let which_args = WhichArgs {
+        applications: args.rest(0)?,
```

Comment:
this would change `which` to let you call it with zero arguments. I think we should require at least one argument

Reply:
Correct. I think this sort of stuff is a common case for a lots of commands.
Maybe we should add a convenience function like: req_at_least(pos, count)?

Reply:
yeah, I like it. Could call it something like `.rest_with_minimum(pos, count)`

Reply:
I have now implement rest_with_minimum and which is using it :)